x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/InstanceTypeService.java

```diff
@@ -27,33 +27,37 @@ private static final class Holder {
         private static final Map<InstanceType, CostEntry> costsPerDatacenter;
 
         static {
-            final Map<Object, Object> objects = new HashMap<>();
+            final Map<Object, Object> objects = new HashMap<>(1024);
```

Comment:
Can you please add a short comment for all pre-sized hashmaps why you picked these values? (so we can adapt them later if needed)

Reply:
Thanks, added the comments and amended the values as we now have Azure data to be counted in.

Ideally, we determine these numbers automatically at build time and the use an exact pre-allocation. Have to look into the gradle build system.

Reply:
I removed the pre-allocation now, see https://github.com/elastic/elasticsearch/pull/104964#discussion_r1473166744